root/usr/bin/discovery-setup

```diff
@@ -54,10 +54,10 @@ unless zips.nil?
       else
         log_msg "TFTP error downloading #{zip}: #{tftp_result}"
       end
-
     rescue
       # We always want to write /etc/default/discovery so don't bomb out here
       log_err "Problem downloading zipfile #{zip}"
+      exit 1
```

Comment:
Save the exit code for later (to exit with after writing the env-files), or remove this comment.